spec/components/custom_wizard/mapper_spec.rb

```diff
@@ -250,4 +276,135 @@
       user: user1
     ).perform).to eq(false)
   end
+
+  context "Output Templating" do
+    it "passes the correct values to the template" do
+      template = "w{step_1_field_1}"
+      mapper = create_template_mapper(template_params, user1)
+      result = mapper.interpolate(
+        template.dup,
+        template: true,
+        user: true,
+        wizard: true,
+        value: true
+      )
+      expect(result).to eq(template_params["step_1_field_1"])
+    end
+
+    it "treats replaced values as string literals" do
+      template = '{{ "w{step_1_field_1}" | size }}'
+      mapper = create_template_mapper(template_params, user1)
+      result = mapper.interpolate(
+        template.dup,
+        template: true,
+        user: true,
+        wizard: true,
+        value: true
+      )
+      expect(result).to eq(template_params["step_1_field_1"].size.to_s)
+    end
+
+    it "allows the wizard values to be used inside conditionals" do
+      template = <<-LIQUID
+        {%- if "w{step_1_field_1}" contains "ello" -%}
+          Correct
+        {%- else -%}
+          Incorrect
+        {%-endif-%}
+      LIQUID
+      mapper = create_template_mapper(template_params, user1)
+      result = mapper.interpolate(
+        template.dup,
+        template: true,
+        user: true,
+        wizard: true,
+        value: true
+      )
+      expect(result).to eq("Correct")
+    end
+
+    it "can access data passed to render method as variable" do
+      template = "{{step_1_field_1.size}}"
+      mapper = create_template_mapper(template_params, user1)
+      result = mapper.interpolate(
+        template.dup,
+        template: true,
+        user: true,
+        wizard: true,
+        value: true
+      )
+      expect(result).to eq(template_params["step_1_field_1"].size.to_s)
+    end
+
+    it "doesn't parse the template when template param is false" do
+      template = <<-LIQUID.strip
+        {{ "w{step_1_field_1}" | size}}
+      LIQUID
+      mapper = create_template_mapper(template_params, user1)
+      result = mapper.interpolate(
+        template.dup,
+        template: false,
+      )
+      expect(result).to eq(template)
+    end
+
+    context "custom filter: 'first_non_empty'" do
+      it "gives first non empty element from list" do
+        template = <<-LIQUID.strip
+          {%- assign entry = "" | first_non_empty: step_1_field_1, step_1_field_2, step_1_field_3 -%}
+          {{ entry }}
+        LIQUID
+        mapper = create_template_mapper(template_params_non_empty, user1)
+        result = mapper.interpolate(
+          template.dup,
+          template: true,
+          user: true,
+          wizard: true,
+          value: true
+        )
+        expect(result).to eq(template_params_non_empty["step_1_field_3"])
+      end
+
+      it "gives first non empty element from list  when multiple non empty values present" do
```

Comment:
@fzngagan There's an additional space in this line. Have you run rubocop on this?

Reply:
Are you sure. Rubocop says no offences detected. Also checked the line on the editor.

Reply:
> from list  when multiple

should be

> from list when multiple

Reply:
@fzngagan You can see it just above

<img width="331" alt="Screenshot at Apr 09 14-58-00" src="https://user-images.githubusercontent.com/5931623/114130473-007dc380-9944-11eb-8fa4-6d7e592f04ab.png">

Reply:
Oh I see. Rubocop would consider it an error if was a trailing whitespace at the end of the line or an indentation issue.

Fixed :)